packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/text/TextLayoutManagerMapBuffer.java

```diff
@@ -378,6 +382,54 @@ private static Layout createLayout(
     return layout;
   }
 
+  public static void adjustSpannableFontToFit(
+    Spannable text,
+    float width,
+    YogaMeasureMode widthYogaMeasureMode,
+    float height,
+    YogaMeasureMode heightYogaMeasureMode,
+    double minimumFontSizeAttr,
+    double maximumFontSizeAttr,
+    int maximumNumberOfLines,
+    boolean includeFontPadding,
+    int textBreakStrategy,
+    int hyphenationFrequency) {
+    BoringLayout.Metrics boring = BoringLayout.isBoring(text, sTextPaintInstance);
+    Layout layout = createLayout(text, boring, width, widthYogaMeasureMode, includeFontPadding, textBreakStrategy, hyphenationFrequency);
+
+    // Find the largest font size used in the spannable to use as a starting point.
+    // Minimum font size is 4pts to match the iOS implementation.
+    int currentFontSize = (int) (Double.isNaN(minimumFontSizeAttr) ? PixelUtil.toPixelFromDIP(4) : minimumFontSizeAttr);
+    ReactAbsoluteSizeSpan[] spans = text.getSpans(0, text.length(), ReactAbsoluteSizeSpan.class);
+    for (ReactAbsoluteSizeSpan span : spans) {
+      currentFontSize = Math.max(currentFontSize, span.getSize());
+    }
+
+    int initialFontSize = currentFontSize;
+    int minimumFontSize = (int) (Double.isNaN(minimumFontSizeAttr) ? PixelUtil.toPixelFromDIP(4) : minimumFontSizeAttr);
```

Comment:
This is practically duplicated from line 402. Could we avoid doing this check twice?

Reply:
Done in https://github.com/facebook/react-native/pull/44075/commits/4451253f2cde47afe666bf60ccb97642960faf33